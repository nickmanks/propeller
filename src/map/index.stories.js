import React from 'react';
import Map from '.';
import Zoom from './controls/zoom';
import Pan from './controls/pan';
import Keyboard from './controls/keyboard';
import TileLayer from './layers/tile';
import {component} from '#src/testing/snapshots';

component('Map')
  .when('standard', ()=> (
    <Map defaultZoom={0} minZoom={-1} maxZoom={3}>
      <Zoom />
      <Pan />
      <Keyboard />
      <TileLayer url={'http://localhost:9000/tiles'} />
    </Map>
  ))
  .when('with zoom', ()=> (
    <Map defaultZoom={0} minZoom={-1} maxZoom={3}>
      <Zoom />
      <TileLayer url={'http://localhost:9000/tiles'} />
    </Map>
  ))
  .when('with pan', ()=> (
    <Map defaultZoom={0} minZoom={-1} maxZoom={3}>
      <Pan />
      <TileLayer url={'http://localhost:9000/tiles'} />
    </Map>
  ))
  .when('with keyboard controls', ()=> (
    <Map defaultZoom={0} minZoom={-1} maxZoom={3}>
      <Keyboard />
      <TileLayer url={'http://localhost:9000/tiles'} />
    </Map>
  ));
