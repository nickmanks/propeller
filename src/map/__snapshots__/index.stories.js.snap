// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`standard 1`] = `
<div
  className="mock-map"
  id="map-container"
>
  <canvas
    className="mock-canvas"
  />
</div>
`;
